code/go/0chain.net/blobbercore/allocation/connection.go

```diff
@@ -0,0 +1,78 @@
+package allocation
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+var (
+	// ConnectionObjCleanInterval start to clean the connectionObjMap
+	ConnectionObjCleanInterval = 10 * time.Minute
+	// ConnectionObjTimout after which connectionObj entry should be invalid
+	ConnectionObjTimeout = 10 * time.Minute
+)
+
+var (
+	connectionObjSizeMap = make(map[string]ConnectionObjSize)
+	connectionObjMutex   sync.Mutex
+)
+
+type ConnectionObjSize struct {
+	Size      int64
+	UpdatedAt time.Time
+}
+
+// GetConnectionObjSize gets the connection size from the memory
+func GetConnectionObjSize(connectionID string) int64 {
+	connectionObjMutex.Lock()
+	defer connectionObjMutex.Unlock()
+	return connectionObjSizeMap[connectionID].Size
+}
+
+// UpdateConnectionObjSize updates the connection size by addSize in memory
+func UpdateConnectionObjSize(connectionID string, addSize int64) {
+	connectionObjMutex.Lock()
+	connectionObjSize := connectionObjSizeMap[connectionID]
+	connectionObjSizeMap[connectionID] = ConnectionObjSize{Size: connectionObjSize.Size + addSize, UpdatedAt: time.Now()}
```

Comment:
Can we have a map of pointers, we are creating a copy for every change

Reply:
@Roshan-Mehta

Reply:
Done